### DISCUSSION TITLE
Problems with ram

### DISCUSSION THREAD

**Discussion by zarMarco** [zbm-dev/zfsbootmenu#544] [Questions]
Hi all, on my system I've one pool, with 3 datasetz one for arch, one Gentoo and one void. Only on arch I had mounted efi and I installed zfsbootmenu on arch. It is perfect but sometimes Gentoo or void kernel loaded read only 450mb of ram, and I have system out of memory. I don't know where is the problem, but yesterday I had install zfsbootmenu on Gentoo too and Gentoo and void went correctly. Today after install of artix I had same problems on artix and Gentoo. solved after install the third zfsbootmenu, on void. I have finish systems where I can install zfsbootmenu 😁.
Someone has the same problem, or can help me?

Root datasets are mounted on /
Home datasets are mounted in legacy and insert on fstab

**Comment by zdykstra** [comment]
How much memory does your system physically have? How much is available in the ZFSBootMenu recovery shell? What kernel(s) are you using?

**Reply by zarMarco** [reply]
I have 16gb of physical ram. I had have this problem on void with kernel 6.5.12 and 6.6.1 but not with 6.6.2 On Gentoo I use gentoo-kernel-bin and I had have problems with 6.6.1, 6.6.2 and with 6.6.3, but now on 6.6.2 goes. On arch I don't have problems

**Comment by zarMarco** [comment]
Today I've same problem also with arch. I just tried to remove and create a new generate-zbm but mount always 450mb of Ram. But currently if boot arch from ZBM install on void I booted correctly. I don't have idea how can I solved this problem.

**Reply by classabbyamp** [reply]
This sounds like a hardware issue, not a ZBM issue

**Reply by zarMarco** [reply]
I don't exclude this. But via emergency shell ZBM I see correct ram

**Reply by zarMarco** [reply]
I think if I have problems with ram, I should have this problem random. But if I have one distro with this problem, always this problem there. I'm going to stop having it maybe with update kernel
